zingolib/src/wallet/transaction_record.rs

```diff
@@ -324,9 +324,39 @@ impl TransactionRecord {
     }
 }
 
+#[cfg(feature = "test-features")]
+pub(crate) mod mocks {
+    use crate::wallet::notes::{SaplingNote, TransparentNote};
+
+    use super::TransactionRecord;
+
+    impl TransactionRecord {
+        #[allow(dead_code)]
+        pub(crate) fn mock() -> Self {
```

Comment:
I think we should follow the builder pattern and maybe you can use this as default?